lib/objects/objectsInMemServerRedis.js

```diff
@@ -606,34 +606,55 @@ class ObjectsInMemoryServer extends ObjectsInMemoryFileDB {
             }
         });
 
+        // handle Redis "SCAN" request for state namespace
+        handler.on('scan', (data, responseId) => {
+            if (!data || data.length < 3) {
+                return handler.sendArray(responseId, ['0', []]);
+            }
+
+            return _handleScanOrKeys(data[2], responseId, true);
+        });
+
         // Handle Redis "KEYS" request for state namespace
         handler.on('keys', (data, responseId) => {
             if (!data || !data.length) {
                 return handler.sendArray(responseId, []);
             }
 
-            const {id, namespace, name, isMeta} = this._normalizeId(data[0]);
+            return _handleScanOrKeys(data[0], responseId);
+        });
+
+        /**
+         * Get keys matching pattern and send it to given responseId, for "SCAN" and "KEYS" - Objects and files supported
+         *
+         * @param {string} pattern - pattern without namespace prefix
+         * @param {number} responseId - Id where response will be sent to
+         * @param {boolean} isScan - if used by "SCAN" this flag should be true
+         * @private
+         */
+        const _handleScanOrKeys = (pattern, responseId, isScan= false) => {
```

Comment:
from my point of veiw (also that we do not really have access procection, a normal class method starting with _ would aoso be possible ... when we use  ES6 classes we should avoid "Inline functions" when possible, or?! (discussion topic)

Reply:
I can also make a class method and pass the handler to it if preferred.